scripts/nm-upgrade.sh

```diff
@@ -0,0 +1,574 @@
+#!/bin/bash
+
+CONFIG_FILE=netmaker.env
+# location of nm-quick.sh (usually `/root`)
+SCRIPT_DIR=$(dirname "$(realpath "$0")")
+CONFIG_PATH="$SCRIPT_DIR/$CONFIG_FILE"
+NM_QUICK_VERSION="0.1.0"
+LATEST=$(curl -s https://api.github.com/repos/gravitl/netmaker/releases/latest | grep "tag_name" | cut -d : -f 2,3 | tr -d [:space:],\")
+
+if [ "$(id -u)" -ne 0 ]; then
+	echo "This script must be run as root"
+	exit 1
+fi
+
+unset INSTALL_TYPE
+unset NETMAKER_BASE_DOMAIN
+
+# usage - displays usage instructions
+usage() {
+	echo "nm-upgrade.sh v$NM_QUICK_VERSION"
+	echo "usage: ./nm-upgrade.sh"
+	exit 1
+}
+
+while getopts v flag; do
+	case "${flag}" in
+	v)
+		usage
+		exit 0
+		;;
+	*)
+		usage
+		exit 0
+		;;
+	esac
+done
+
+# print_logo - prints the netmaker logo
+print_logo() {
+	cat <<"EOF"
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+                                                                                         
+ __   __     ______     ______   __    __     ______     __  __     ______     ______    
+/\ "-.\ \   /\  ___\   /\__  _\ /\ "-./  \   /\  __ \   /\ \/ /    /\  ___\   /\  == \   
+\ \ \-.  \  \ \  __\   \/_/\ \/ \ \ \-./\ \  \ \  __ \  \ \  _"-.  \ \  __\   \ \  __<   
+ \ \_\\"\_\  \ \_____\    \ \_\  \ \_\ \ \_\  \ \_\ \_\  \ \_\ \_\  \ \_____\  \ \_\ \_\ 
+  \/_/ \/_/   \/_____/     \/_/   \/_/  \/_/   \/_/\/_/   \/_/\/_/   \/_____/   \/_/ /_/ 
+                                                                                                                                                                                                 
+
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+EOF
+}
+
+# set_buildinfo - sets the information based on script input for how the installation should be run
+set_buildinfo() {
+
+	MASTERKEY=$(grep MASTER_KEY docker-compose.yml | awk '{print $2;}' | tr -d '"')
+	EMAIL=$(grep email Caddyfile | awk '{print $2;}' | tr -d '"')
+	BROKER=$(grep SERVER_NAME docker-compose.yml | awk '{print $2;}' | tr -d '"')
```

Comment:
@mattkasun @theguy951357 @abhishek9686 



What is the expected value of `BROKER=$(grep SERVER_NAME docker-compose.yml | awk '{print $2;}' | tr -d '"')`

For me, it returns `SERVER_NAME=${NM_DOMAIN}` and result an errors when get the certificates


```
root@nm:~# ./nm-upgrade.sh 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
                                                                                         
 __   __     ______     ______   __    __     ______     __  __     ______     ______    
/\ "-.\ \   /\  ___\   /\__  _\ /\ "-./  \   /\  __ \   /\ \/ /    /\  ___\   /\  == \   
\ \ \-.  \  \ \  __\   \/_/\ \/ \ \ \-./\ \  \ \  __ \  \ \  _"-.  \ \  __\   \ \  __<   
 \ \_\\"\_\  \ \_____\    \ \_\  \ \_\ \ \_\  \ \_\ \_\  \ \_\ \_\  \ \_____\  \ \_\ \_\ 
  \/_/ \/_/   \/_____/     \/_/   \/_/  \/_/   \/_/\/_/   \/_/\/_/   \/_____/   \/_/ /_/ 
                                                                                                                                                                                                 

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Using config: /root/netmaker.env
-----------------------------------------------------
Would you like to install Netmaker Community Edition (CE), or Netmaker Enterprise Edition (EE)?
EE will require you to create an account at https://app.netmaker.io
-----------------------------------------------------
1) Community Edition
2) Enterprise Edition
#? 2
installing Netmaker EE
-----------Build Options-----------------------------
    EE or CE: ee
   Version: v0.20.6
   Installer: v0.1.0
-----------------------------------------------------
checking dependencies...
Hit:1 http://ubuntu.mirror.constant.com jammy InRelease
Hit:2 https://download.docker.com/linux/ubuntu jammy InRelease                                                                                         
Hit:3 http://ubuntu.mirror.constant.com jammy-updates InRelease                                                                                        
Get:4 http://ubuntu.mirror.constant.com jammy-backports InRelease [109 kB]                           
Hit:5 https://apprepo.vultr.com/ubuntu universal InRelease                                                      
Hit:6 http://archive.ubuntu.com/ubuntu jammy InRelease                                                          
Get:7 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [119 kB]
Hit:8 http://ubuntu.mirror.constant.com jammy-security InRelease                                                                                                                            
Get:9 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [109 kB]                                                                                                                   
Get:10 http://archive.ubuntu.com/ubuntu jammy-security InRelease [110 kB]                                                                                                                   
Fetched 446 kB in 8s (53.8 kB/s)                                                                                                                                                            
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
63 packages can be upgraded. Run 'apt list --upgradable' to see them.
     git is installed
     wireguard is installed
     wireguard-tools is installed
     dnsutils is installed
     jq is installed
     docker.io is not installed. Attempting install.
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:

The following packages have unmet dependencies:
 containerd.io : Conflicts: containerd
E: Error, pkgProblemResolver::Resolve generated breaks, this may be caused by held packages.
   FAILED TO INSTALL docker.io
   This may break functionality.
     docker-compose is installed
     grep is installed
     gawk is installed
-----------------------------------------------------
dependency check complete
-----------------------------------------------------
-----------------------------------------------------
The following subdomains will be used:
          dashboard.SERVER_NAME=${NM_DOMAIN}
                api.SERVER_NAME=${NM_DOMAIN}
             broker.SERVER_NAME=${NM_DOMAIN}
               turn.SERVER_NAME=${NM_DOMAIN}
            turnapi.SERVER_NAME=${NM_DOMAIN}
         prometheus.SERVER_NAME=${NM_DOMAIN}
  netmaker-exporter.SERVER_NAME=${NM_DOMAIN}
            grafana.SERVER_NAME=${NM_DOMAIN}
-----------------------------------------------------
-----------------------------------------------------
Provide Details for EE installation:
    1. Log into https://app.netmaker.io
    2. follow instructions to get a license at: https://docs.netmaker.io/ee/ee-setup.html
    3. Retrieve License and Tenant ID
    4. note email address
-----------------------------------------------------
License Key: ***********************
Tenant ID: ***********************
using default username/random pass for MQ
using default username/random pass for TURN
-----------------------------------------------------------------
                SETUP ARGUMENTS
-----------------------------------------------------------------
        domain: SERVER_NAME=${NM_DOMAIN}
         email: ***********************@gmail.com
     public ip: ***.***.***.***
       license: ***********************
    account id: ***********************
-----------------------------------------------------------------
Confirm Settings for Installation
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Does everything look right? [y/n]: y
Stopping all containers...
generating netclient configuration files
{"time":"2023-08-28T16:16:56.157767822Z","level":"INFO","source":"root.go 91}","msg":"creating config dir"}
{"time":"2023-08-28T16:16:56.15870603Z","level":"INFO","source":"root.go 95}","msg":"retrieving legacy nodes"}
{"time":"2023-08-28T16:16:56.163801576Z","level":"WARN","source":"root.go 149}","msg":"unmarhal node","error":"json: cannot unmarshal bool into Go struct field LegacyNode.connected of type string"}
{"time":"2023-08-28T16:16:56.163839299Z","level":"WARN","source":"root.go 149}","msg":"unmarhal node","error":"json: cannot unmarshal bool into Go struct field LegacyNode.connected of type string"}
{"time":"2023-08-28T16:16:56.163859077Z","level":"WARN","source":"root.go 149}","msg":"unmarhal node","error":"json: cannot unmarshal bool into Go struct field LegacyNode.connected of type string"}
-----------------------------------------------------------------
Beginning installation...
-----------------------------------------------------------------
. . .
. . .
. . .
Pulling config files...
Saving the config to /root/netmaker.env
Setting up SSL certificates...
fetch https://dl-cdn.alpinelinux.org/alpine/v3.16/main/x86_64/APKINDEX.tar.gz
fetch https://dl-cdn.alpinelinux.org/alpine/v3.16/community/x86_64/APKINDEX.tar.gz
v3.16.7-45-ga175333e49e [https://dl-cdn.alpinelinux.org/alpine/v3.16/main]
v3.16.7-44-g4bb68229aaf [https://dl-cdn.alpinelinux.org/alpine/v3.16/community]
OK: 17048 distinct packages available
(1/5) Installing bash (5.1.16-r2)
Executing bash-5.1.16-r2.post-install
(2/5) Installing brotli-libs (1.0.9-r6)
(3/5) Installing nghttp2-libs (1.47.0-r1)
(4/5) Installing libcurl (8.2.1-r0)
(5/5) Installing curl (8.2.1-r0)
Executing busybox-1.35.0-r17.trigger
OK: 29 MiB in 46 packages
certonly --standalone --non-interactive --expand --agree-tos -m ****@gmail.com -d api.SERVER_NAME= -d broker.SERVER_NAME= -d dashboard.SERVER_NAME= -d turn.SERVER_NAME= -d turnapi.SERVER_NAME= -d netmaker-exporter.SERVER_NAME= -d grafana.SERVER_NAME= -d prometheus.SERVER_NAME=
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Requesting a certificate for api.server_name= and 7 more domains
An unexpected error occurred:
Error creating new order :: Cannot issue for "api.server_name=": Domain name contains an invalid character (and 7 more problems. Refer to sub-problems for more information.)
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Requesting a certificate for api.server_name= and 7 more domains
An unexpected error occurred:
Error creating new order :: Cannot issue for "api.server_name=": Domain name contains an invalid character (and 7 more problems. Refer to sub-problems for more information.)
Ask for help or search for solutions at https://community.letsencrypt.org. See the logfile /var/log/letsencrypt/letsencrypt.log or re-run Certbot with -v for more details.
Missing file: /root/letsencrypt/live/api.SERVER_NAME=/fullchain.pem
SSL certificates failed
```